From 8d5c8bdd7d36d86997cac2790c82007720829264 Mon Sep 17 00:00:00 2001
From: Harshitha Prem <hprem@qti.qualcomm.com>
Date: Thu, 11 Dec 2025 14:17:16 +0530
Subject: [PATCH 1/2] QSDK:CP: macsec: revert max rate information

This patch reverts the commit of b00-034-hostapd-Add-max-rate-information-into-STATUS-and-STA.patch

Signed-off-by: Harshitha Prem <hprem@qti.qualcomm.com>
---
 hostapd/ctrl_iface.c   |   2 +-
 src/ap/ctrl_iface_ap.c | 113 -----------------------------------------
 src/ap/ctrl_iface_ap.h |   2 +-
 src/ap/sta_info.h      |   4 --
 4 files changed, 2 insertions(+), 119 deletions(-)

diff --git a/hostapd/ctrl_iface.c b/hostapd/ctrl_iface.c
index 89f9336..6f743ee 100644
--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -3261,7 +3261,7 @@ static int hostapd_ctrl_iface_color_change(struct hostapd_iface *iface,
 #endif /* CONFIG_IEEE80211AX */
 
 
-u8 hostapd_maxnss(struct hostapd_data *hapd, struct sta_info *sta)
+static u8 hostapd_maxnss(struct hostapd_data *hapd, struct sta_info *sta)
 {
 	u8 nss = 0;
 	u8 rx_nss = 1;
diff --git a/src/ap/ctrl_iface_ap.c b/src/ap/ctrl_iface_ap.c
index c9d386a..bdf7f7d 100644
--- a/src/ap/ctrl_iface_ap.c
+++ b/src/ap/ctrl_iface_ap.c
@@ -111,47 +111,6 @@ static int hostapd_get_sta_conn_time(struct sta_info *sta,
 	return ret;
 }
 
-static u8 hostapd_htmaxmcs(const u8 *mcs_set)
-{
-	u8 rates[WLAN_SUPP_RATES_MAX];
-	u8 i;
-	u8 j = -1;
-
-	for (i = 0; i < WLAN_SUPP_HT_RATES_MAX; i++) {
-		if (j == WLAN_SUPP_RATES_MAX) {
-			wpa_printf(MSG_INFO,
-				   "HT extended rate set too large; using only %u rates",
-				    j);
-			break;
-		}
-		if (mcs_set[i / 8] & (1 << (i % 8)))
-			rates[++j] = i;
-	}
-	if (j > -1)
-		return rates[j];
-
-	return 0;
-}
-
-
-static u8 hostapd_vhtmaxmcs(u16 rx_vht_mcs_map, u16 tx_vht_mcs_map)
-{
-	u8 rx_max_mcs, tx_max_mcs, max_mcs;
-
-	if (rx_vht_mcs_map && tx_vht_mcs_map) {
-		/* Refer to IEEE P802.11ac/D7.0 Figure 8-401bs
-		 * for VHT MCS Map definition
-		 */
-		rx_max_mcs = rx_vht_mcs_map & 0x03;
-		tx_max_mcs = tx_vht_mcs_map & 0x03;
-		max_mcs = rx_max_mcs < tx_max_mcs ? rx_max_mcs : tx_max_mcs;
-		if (max_mcs < 0x03)
-			return 7 + max_mcs;
-	}
-
-	return 0;
-}
-
 
 static int hostapd_get_sta_info(struct hostapd_data *hapd,
 				struct sta_info *sta,
@@ -673,39 +632,6 @@ static int hostapd_ctrl_iface_sta_mib(struct hostapd_data *hapd,
 	}
 #endif /* CONFIG_IEEE80211BE */
 
-	ret = os_snprintf(buf + len, buflen - len, "max_nss=%u\n",
-			hostapd_maxnss(hapd, sta));
-	if (!os_snprintf_error(buflen - len, ret))
-		len += ret;
-
-#ifdef CONFIG_IEEE80211AC
-	if ((sta->flags & WLAN_STA_VHT) && sta->vht_capabilities) {
-		u8 vht_maxmcs = hostapd_vhtmaxmcs(
-				le_to_host16(sta->vht_capabilities->
-					vht_supported_mcs_set.rx_map),
-				le_to_host16(sta->vht_capabilities->
-					vht_supported_mcs_set.tx_map));
-		ret = os_snprintf(buf + len, buflen - len, "max_vhtmcs=%u\n",
-				vht_maxmcs);
-		if (!os_snprintf_error(buflen - len, ret))
-			len += ret;
-	}
-#endif /* CONFIG_IEEE80211AC */
-
-#ifdef CONFIG_IEEE80211N
-	if ((sta->flags & (WLAN_STA_HT | WLAN_STA_VHT)) == WLAN_STA_HT &&
-			sta->ht_capabilities) {
-		u8 ht_maxmcs;
-
-		ht_maxmcs = hostapd_htmaxmcs(sta->ht_capabilities->
-				supported_mcs_set);
-		ret = os_snprintf(buf + len, buflen - len, "max_mcs=%u\n",
-				ht_maxmcs);
-		if (!os_snprintf_error(buflen - len, ret))
-			len += ret;
-	}
-#endif /* CONFIG_IEEE80211N */
-
 #ifdef CONFIG_IEEE80211BE
 	if (sta->mld_info.mld_sta == true) {
 		size_t link_type;
@@ -1268,20 +1194,6 @@ int hostapd_ctrl_iface_status(struct hostapd_data *hapd, char *buf,
 		if (os_snprintf_error(buflen - len, ret))
 			return len;
 		len += ret;
-
-		if (mode) {
-			u16 rxmap = mode->vht_mcs_set[0] |
-				(mode->vht_mcs_set[1] << 8);
-			u16 txmap = mode->vht_mcs_set[4] |
-				(mode->vht_mcs_set[5] << 8);
-
-			ret = os_snprintf(buf + len, buflen - len,
-					"vht_max_mcs=%u\n",
-					hostapd_vhtmaxmcs(rxmap, txmap));
-			if (os_snprintf_error(buflen - len, ret))
-				return len;
-			len += ret;
-		}
 	}
 
 	if (iface->conf->ieee80211n && !hapd->conf->disable_11n) {
@@ -1316,33 +1228,8 @@ int hostapd_ctrl_iface_status(struct hostapd_data *hapd, char *buf,
 		if (os_snprintf_error(buflen - len, ret))
 			return len;
 		len += ret;
-
-		if (mode && iface->conf->ieee80211n) {
-			ret = os_snprintf(buf + len, buflen - len,
-					"max_mcs=%u\n",
-					hostapd_htmaxmcs(mode->mcs_set));
-			if (os_snprintf_error(buflen - len, ret))
-				return len;
-			len += ret;
-		}
 	}
 
-	if (mode && mode->rates && mode->num_rates &&
-			mode->num_rates <= WLAN_SUPP_RATES_MAX) {
-		ret = os_snprintf(buf + len, buflen - len,
-				"max_rate=%u\n",
-				mode->rates[mode->num_rates - 1]);
-		if (os_snprintf_error(buflen - len, ret))
-			return len;
-		len += ret;
-	}
-
-	ret = os_snprintf(buf + len, buflen - len, "max_nss=%u\n",
-			hostapd_maxnss(hapd, NULL));
-	if (os_snprintf_error(buflen - len, ret))
-		return len;
-	len += ret;
-
 	for (j = 0; mode && j < mode->num_channels; j++) {
 		if (mode->channels[j].freq == iface->freq) {
 			ret = os_snprintf(buf + len, buflen - len,
diff --git a/src/ap/ctrl_iface_ap.h b/src/ap/ctrl_iface_ap.h
index 492eb55..1552bea 100644
--- a/src/ap/ctrl_iface_ap.h
+++ b/src/ap/ctrl_iface_ap.h
@@ -68,5 +68,5 @@ int hostapd_ctrl_iface_acl_show_mac(struct mac_acl_entry *acl, int num,
 				    char *buf, size_t buflen);
 int hostapd_disassoc_accept_mac(struct hostapd_data *hapd);
 int hostapd_disassoc_deny_mac(struct hostapd_data *hapd);
-u8 hostapd_maxnss(struct hostapd_data *hapd, struct sta_info *sta);
+
 #endif /* CTRL_IFACE_AP_H */
diff --git a/src/ap/sta_info.h b/src/ap/sta_info.h
index b482ed6..e58208f 100644
--- a/src/ap/sta_info.h
+++ b/src/ap/sta_info.h
@@ -59,10 +59,6 @@
 /* wired mac authentication bypass sta, non-802.1x capable */
 #define WIRED_STA_MAB BIT(2)
 
-/* Maximum number of supported rates (from both Supported Rates and Extended
- * Supported Rates IEs). */
-#define WLAN_SUPP_RATES_MAX 32
-#define WLAN_SUPP_HT_RATES_MAX 77
 
 struct hostapd_data;
 
-- 
2.34.1

